src/main/java/edu/harvard/iq/dataverse/util/SignpostingResources.java

```diff
@@ -0,0 +1,321 @@
+import javax.json.JsonObjectBuilder;
+import javax.json.JsonObject;
+import javax.json.JsonReader;
+import java.io.IOException;
+import java.io.StringReader;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Objects;
+import java.util.logging.Logger;
+
+import static edu.harvard.iq.dataverse.util.json.NullSafeJsonBuilder.jsonObjectBuilder;
+
+public class SignpostingResources {
+    private static final Logger logger = Logger.getLogger(SignpostingResources.class.getCanonicalName());
+    SystemConfig systemConfig;
+    DatasetVersion workingDatasetVersion;
+    JsonObject describedByJsonObj;
+    Boolean useDefaultFileType;
+    String defaultFileTypeValue;
+    int maxAuthors;
+    int maxItems;
+
+    public SignpostingResources(SystemConfig systemConfig, DatasetVersion workingDatasetVersion, String jsonSetting) {
+        this.systemConfig = systemConfig;
+        this.workingDatasetVersion = workingDatasetVersion;
+        if (jsonSetting == null) {
+            jsonSetting = BundleUtil.getStringFromBundle("signposting.configuration.SignpostingConf");
+        }
+        JsonReader jsonReader = Json.createReader(new StringReader(jsonSetting));
+        JsonObject spJsonSetting = jsonReader.readObject();
+        jsonReader.close();
+        describedByJsonObj = spJsonSetting.getJsonObject("describedby");
+        useDefaultFileType = spJsonSetting.getBoolean("useDefaultFileType", true);
+        defaultFileTypeValue = spJsonSetting.getString("defaultFileTypeValue", "https://schema.org/Dataset");
+        maxAuthors = spJsonSetting.getInt("maxAuthors", 5);
+        maxItems = spJsonSetting.getInt("maxItems", 5);
+    }
+
+    /**
+     * Get Authors as string
+     * For example:
+     * if author has VIAF
+     * Link: <http://viaf.org/viaf/:id/>; rel="author"
+     *
+     * @param datasetAuthors list of all DatasetAuthor object
+     * @return all the non empty author links in a string
+     */
+    private String getAuthorsAsString(List<DatasetAuthor> datasetAuthors) {
+        String singleAuthorString;
+        String identifierSchema = "";
+        int visibleAuthorCounter = 0;
+//        if (datasetAuthors.size() > maxAuthors) {return "";}
+        for (DatasetAuthor da : datasetAuthors) {
+            logger.info(String.format(
+                    "idtype: %s; idvalue: %s, affiliation: %s; identifierUrl: %s",
+                    da.getIdType(),
+                    da.getIdValue(),
+                    da.getAffiliation(),
+                    da.getIdentifierAsUrl()
+            ));
+
+            String authorURL = "";
+            authorURL = getAuthorUrl(da);
+            if (!Objects.equals(authorURL, "")) {
+                visibleAuthorCounter++;
+                // return empty if number of visible author more than max allowed
+                if (visibleAuthorCounter >= maxAuthors) return "";
+                singleAuthorString = "<" + authorURL + ">;rel=\"author\"";
```

Comment:
I'm not sure why but I'm getting null for authorURL like this (I'm on 5fe6a8a8ee):

```
Link: <null>;rel="author", <https://doi.org/...
```

This is just a local dataset in dev. Here's the curl command:

```
curl -s -i http://localhost:8080/dataset.xhtml?persistentId=doi:10.5072/FK2/KQRKB8
```